cmd/juju/application/addrelation.go

```diff
@@ -146,18 +150,31 @@ func (c *addRelationCommand) Run(ctx *cmd.Context) error {
 }
 
 func (c *addRelationCommand) maybeConsumeOffer(targetClient applicationAddRelationAPI) error {
-	sourceClient, err := c.getOffersAPI()
+	sourceClient, err := c.getOffersAPI(c.remoteEndpoint)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	defer sourceClient.Close()
 
+	// The source has been used to get the correct client api,
+	// so now can be discarded as the api rejects attempts to
+	// connect with a non-empty source.
+	sourceController := c.remoteEndpoint.Source
+	c.remoteEndpoint.Source = ""
```

Comment:
Rather than mutating the official one, might be nicer to have a copy with a blank Source? Thinking about confusion during debugging later on. Or what about `.AsLocal()` which would omit the Source for communication with the offering controller.
``` lang=go
consumeDetails, err := sourceClient.GetConsumeDetails(c.remoteEndpoint.AsLocal().String())
```